peps/pep-0771.rst

```diff
@@ -202,34 +260,84 @@ extras.
-packages will remain fully backward-compatible with older packaging tools - with
-the only difference that the default extras will not be installed automatically
-when older packaging tools are used.
+packages will remain installable with older packaging tools -- with the main
+difference being that the default extras will not be installed automatically
+when older packaging tools are used. As described in `How to teach this`_,
+package authors need to carefully evaluate when and how they adopt
+the default extra feature depending on their user base, as some actions (such as
+moving a required dependency to a default extra) will likely result in breakage
+for users if a significant fraction of them are still using older package
+installers that do not support default extras. In this sense, package authors
+should be aware that this feature, if used in certain ways, can cause
+backward-compatibility issues for users, and they are thus responsible for
+ensuring that they minimize the impact to users.
+
+Packaging-related tools
+-----------------------
+
+The most significant backward-compatibility aspect is related to assumptions
+pakaging tools make about extras -- specifically, this PEP changes the
+assumption that extras are no longer exclusively additive in terms of adding
+dependencies to the dependency tree, and specifying some extras can result in
+fewer dependencies being installed.
+
+A specific example of change in behavior can be seen with `pip`_: consider a
+package ``package`` which has a required dependency of ``numpy``, a (default)
+extra called ``recommended`` which includes ``scipy``, and a ``minimal`` extra
+which does not contain any dependencies. If a user installs ``package[minimal]``,
+only ``package`` and ``numpy`` will be installed. If a user then does:
+
+.. code-block:: console
+
+    $ pip freeze > requirements.txt
+
+then ``requirements.txt`` will contain e.g.::
+
+    package==1.0.2
+    numpy==2.1.0
+
+However, if the user then installs the requirements from this file using:
+
+.. code-block:: console
 
-The only conceptual backward-compatibility issue to consider is the fact that
-this PEP changes extras to no longer be strictly additive, in that specifying
-an extra such as ``minimal`` could result in fewer packages being installed.
+    $ pip install -r requirements.txt
+
+then pip will install ``package`` (which will include the default extra) as well
+as ``numpy``, so the final environment will contain ``scipy``. A solution in this
+specific case is for the user to pass ``--no-deps`` to ``pip install`` to avoid
+resolving the dependency tree, but the point here is to illustrate that there
+may be changes in behavior in packaging tools due to the change in the
+assumption about what impact an extra can have.
 
 Security Implications
 =====================
 
 There are no known security implications for this PEP.
 
-How to teach this
+Examples of usage
 =================
```

Comment:
Please stick to the suggested sections in [PEP 12](https://peps.python.org/pep-0012/#suggested-sections). Perhaps you could add "examples of usage" as a sub-section, but "how to teach this" is broader than simply stating examples.
```suggestion
How to teach this
=================
```